lib/friendly_shipping/services/ups.rb

```diff
@@ -0,0 +1,69 @@
+require 'dry/monads/result'
+require 'friendly_shipping/services/ups/client'
+require 'friendly_shipping/services/ups/serialize_access_request'
+require 'friendly_shipping/services/ups/serialize_rating_service_selection_request'
+require 'friendly_shipping/services/ups/parse_rate_response'
+require 'friendly_shipping/services/ups/shipping_methods'
+
+module FriendlyShipping
+  module Services
+    class Ups
+      include Dry::Monads::Result::Mixin
+
+      attr_reader :test, :key, :login, :password, :client
+
+      CARRIER = FriendlyShipping::Carrier.new(
+        id: 'ups',
+        name: 'United Parcel Service',
+        code: 'ups',
+        shipping_methods: SHIPPING_METHODS
+      )
+
+      TEST_URL = 'https://wwwcie.ups.com'
+      LIVE_URL = 'https://onlinetools.ups.com'
+
+      RESOURCES = {
+        rates: '/ups.app/xml/Rate'
+      }
+
+      def initialize(key:, login:, password:, test: true, client: self.class::Client)
+        @key = key
+        @login = login
+        @password = password
+        @test = test
+        @client = client
+      end
+
+      def carriers
+        Success([CARRIER])
+      end
+
+      def rate_estimates(shipment, carriers)
+        rate_request_xml = self.class::SerializeRatingServiceSelectionRequest.(shipment: shipment)
+        url = base_url + RESOURCES[:rates]
+        request = FriendlyShipping::Request.new(
+          url: url,
+          body: access_request_xml + rate_request_xml
+        )
+
+        client.post(request).bind do |response|
+          self.class::ParseRateResponse.(response: response, request: request, shipment: shipment)
+        end
+      end
+
+      private
+
+      def access_request_xml
+        self.class::SerializeAccessRequest.(key: key, login: login, password: password)
```

Comment:
Same here as well, I think it helps to either write `FriendlyShipping::Services::Ups::SerializeAccessRequest` or the short version `SerializeAccessRequest` using `self.class::` could be misleading.

Also I agree with Rubocop that the `.call()` shortcut `.()` can be a bit confusing as well.

Reply:
I'll add Rubocop shortly.